cassandra-analytics-core/src/main/java/org/apache/cassandra/spark/bulkwriter/cloudstorage/ImportCompletionCoordinator.java

```diff
@@ -362,7 +383,7 @@ private void handleFailedSliceInstance(RingInstance instance,
         catch (RuntimeException rte)
         {
             // record the first failure and cancel queued futures.
-            firstFailure.completeExceptionally(rte);
+            setImportFailure(new ImportFailedException(rte));
```

Comment:
can we use propagate here?
```suggestion
            setImportFailure(ImportFailedException.propagate(rte));
```